### PR TITLE
Search cache for urls matching the regex input

### PR DESCRIPTION
**Pull request by persiaAziz**
Searches and lists the urls that match with the input regex url

### CODE REVIEW

File: lib/ts/BufferWriter.h [hunk 1]
```diff
@@ -259,10 +259,10 @@ class FixedBufferWriter : public BufferWriter
     const size_t newSize = _attempted + length;
 
     if (newSize <= _capacity) {
-      ink_assert(_buf != nullptr); // make clang-analyzer happy.
+      // ink_assert(_buf != nullptr); // make clang-analyzer happy.
```
Comment (persiaAziz) [inline lib/ts/BufferWriter.h]: this is a temporary fix to make cache tool work.

Comment (SolidWallOfCode) [reply lib/ts/BufferWriter.h]: Although it breaks clang-analyzer.

**Comment by persiaAziz** [comment]
making a new PR

**Comment by zwoop** [comment]
Remember to remove the Projects and Milestone on PRs that you close (but don't merge).

**Pull request closed without merging**